src/main/java/com/pokegoapi/main/RequestHandler.java

```diff
@@ -129,8 +133,21 @@ public void sendServerRequests(ServerRequest... serverRequests) throws RemoteSer
 				lastAuth = responseEnvelop.getAuthTicket();
 			}
 
-			if (responseEnvelop.getStatusCode() == 102) {
-				throw new LoginFailedException();
+			if (responseEnvelop.getStatusCode() == 102 && GoogleLoginSecrets.refresh_token != null) {
```

Comment:
Ya the login code/logic could do with some refactoring but I dont think i'd be able to do it now. I really wanted to implement this feature as it makes google auth kinda useless without it.

My ideal API design would be to pass
an implementation of Login to PokemonGo and let the implementation handle tokens transparently.
RequestManager should just as the PokemonGo object for AuthInfo when it wants to do a call.